n3fit/src/n3fit/performfit.py

```diff
@@ -163,29 +163,21 @@ def performfit(
         replicas, replica_experiments, nnseeds = zip(*replicas_nnseed_fitting_data_dict)
         # Parse the experiments so that the output data contain information for all replicas
         # as the only different from replica to replica is the experimental training/validation data
-        all_experiments = copy.deepcopy(replica_experiments[0])
-        for i_exp in range(len(all_experiments)):
-            training_data = []
-            validation_data = []
-            for i_rep in range(n_models):
-                training_data.append(replica_experiments[i_rep][i_exp]['expdata'])
-                validation_data.append(replica_experiments[i_rep][i_exp]['expdata_vl'])
-            all_experiments[i_exp]['expdata'] = np.concatenate(training_data, axis=0)
-            all_experiments[i_exp]['expdata_vl'] = np.concatenate(validation_data, axis=0)
         log.info(
             "Starting parallel fits from replica %d to %d", replicas[0], replicas[0] + n_models - 1
         )
-        replicas_info = [(replicas, all_experiments, nnseeds)]
+        replicas_info = [(replicas, replica_experiments, nnseeds)]
     else:
+        # Cases 1 and 2 above are a special case of 3 where the replica idx and the seed should
+        # be a list of just one element
         replicas_info = replicas_nnseed_fitting_data_dict
+        for i, info_tuple in enumerate(replicas_info):
+            replica_idxs = info_tuple[0]
+            nnseeds = info_tuple[2]
+            replicas_info[i] = (tuple([replica_idxs]), [info_tuple[1]], tuple([nnseeds]))
```

Comment:
I'm very confused about this. Why does element 1 not get its own name and doesn't become a tuple?

So these are all lists of 1 elements, but you have one list per replica I guess? Why is not coming everything as a list? Usually it should be like that if the other two are lists of many elements.

Reply:
I agree it's not very readable code @scarlehoff, I made the filling of the replica_info data structure  much more straightforward in commit 66f10689589342177a9a33e621f1318710ed11d1

Reply:
I thought it was still a bit hard to read, I simplified it further in [bc76939](https://github.com/NNPDF/nnpdf/pull/1788/commits/bc76939f2ec25891ac07e3a35fa5d662c35be7d4).